### PR TITLE
server: fix binary numeric type overflow

### PR DESCRIPTION
**Pull request by jackysp**
## What have you changed? (mandatory)
When using jdbc to setInt(-1) to an int type column, it will cause an overflow error.

## What are the type of the changes (mandatory)?
Bug fix.

## How has this PR been tested (mandatory)?
Integration tests.

PTAL @coocood @lysu

### CODE REVIEW

File: server/conn_stmt.go [hunk 1]
```diff
@@ -273,9 +273,9 @@ func parseStmtArgs(args []interface{}, boundParams [][]byte, nullBitmap, paramTy
 			}
 
 			if isUnsigned {
```
Comment (jackysp) [inline server/conn_stmt.go]: Actually, I think binary int does not have an unsigned flag.

File: server/conn_stmt.go [hunk 2]
```diff
@@ -329,7 +329,7 @@ func parseStmtArgs(args []interface{}, boundParams [][]byte, nullBitmap, paramTy
 				return
 			}
 
-			args[i] = float64(math.Float32frombits(binary.LittleEndian.Uint32(paramValues[pos : pos+4])))
+			args[i] = math.Float32frombits(binary.LittleEndian.Uint32(paramValues[pos : pos+4]))
```
Comment (jackysp) [inline server/conn_stmt.go]: Don't know why there is a cast here, either.

**Comment by jackysp** [comment]
Maybe unit tests from MySQL official jdbc are usefully to cover the binary type code.

File: server/conn_stmt.go [hunk 3]
```diff
@@ -273,9 +273,9 @@ func parseStmtArgs(args []interface{}, boundParams [][]byte, nullBitmap, paramTy
 			}
 
 			if isUnsigned {
-				args[i] = uint64(paramValues[pos])
+				args[i] = uint8(paramValues[pos])
```
Comment (shenli) [inline server/conn_stmt.go]: Why use uint8?

**Comment by shenli** [comment]
Where is the integration test.

**Comment by coocood** [comment]
LGTM

**Comment by jackysp** [comment]
Unit test cases added. But still DNM. @shenli @lysu

File: server/conn_stmt.go (hunk 2 above)
Comment (zimulala) [inline server/conn_stmt.go]: Do we need to add a test case for this type?

**Comment by jackysp** [comment]
PTAL @zimulala

**Pull request merged**